llm-finetune/train.py

```diff
@@ -574,17 +591,35 @@ def get_torch_dtype(training_arguments: HFTrainingArguments):
     return torch_dtype
 
 
-def get_model(model_source: str, training_arguments: HFTrainingArguments):
+def get_model(model_source: str, training_arguments: HFTrainingArguments, other_arguments: OtherArguments,):
     # TODO (chiragjn): Should we pass a torch_dtype here?
     logger.info("Loading model...")
-    model = AutoModelForCausalLM.from_pretrained(
-        model_source,
-        trust_remote_code=True,
-        use_cache=False if training_arguments.gradient_checkpointing else True,
-        torch_dtype=get_torch_dtype(training_arguments),
-    )
+    if other_arguments.use_qlora:
+        bnb_config = BitsAndBytesConfig(
+            load_in_4bit=other_arguments.qlora_bit_length == 4,
+            load_in_8bit=other_arguments.qlora_bit_length == 8,
+            bnb_4bit_compute_dtype=get_torch_dtype(training_arguments=training_arguments),
+            bnb_4bit_use_double_quant=other_arguments.use_double_quant,
+            bnb_4bit_quant_type=other_arguments.bnb_4bit_quant_type,
+        )
+        model = AutoModelForCausalLM.from_pretrained(
+            model_source,
+            trust_remote_code=True,
+            use_cache=False if training_arguments.gradient_checkpointing else True,
+            quantization_config=bnb_config,
+        )
+        model = prepare_model_for_kbit_training(model)
```

Comment:
this takes another argument
```suggestion
        model = prepare_model_for_kbit_training(model, use_gradient_checkpointing=training_arguments.gradient_checkpointing)
```